airflow/hooks/webhdfs_hook.py

```diff
@@ -37,8 +38,9 @@ def get_conn(self):
                 if _kerberos_security_mode:
                   client = KerberosClient(connection_str)
                 else:
-                  client = InsecureClient(connection_str)
-                client.content('/')
+                  proxy_user = self.proxy_user or nn.login
+                  client = InsecureClient(connection_str, user=proxy_user)
+                client.status('/')
```

Comment:
As I understand, the content method is called to check if the namenode is valid, but it fails also on current airflow, if the task is run as any user != hdfs, while the status method will not, so I think that calling status is better for checking the namenode validity.
The other change is for adding the "doAs" also to the hdfs client.